account_loan/model/account_loan.py

```diff
@@ -390,18 +390,19 @@ def check_long_term_principal_amount(self):
         """
         lines = self.line_ids.filtered(lambda r: not r.move_ids)
         amount = 0
-        final_sequence = min(lines.mapped('sequence'))
-        for line in lines.sorted('sequence', reverse=True):
-            date = datetime.strptime(
-                line.date, DF).date() + relativedelta(months=12)
-            if self.state == 'draft' or line.sequence != final_sequence:
-                line.long_term_pending_principal_amount = sum(
-                    self.line_ids.filtered(
-                        lambda r: datetime.strptime(r.date, DF).date() >= date
-                    ).mapped('principal_amount'))
-            line.long_term_principal_amount = (
-                line.long_term_pending_principal_amount - amount)
-            amount = line.long_term_pending_principal_amount
+        if lines:
```

Comment:
for reducing the diff, do:
```suggestion
        if not lines:
            return
```